test/ProblemDetailsMiddlewareTest.php

```diff
@@ -45,60 +42,38 @@ public function acceptHeaders() : array
     public function testSuccessfulDelegationReturnsDelegateResponse() : void
     {
         $response = $this->prophesize(ResponseInterface::class);
-        $delegate = $this->prophesize(DelegateInterface::class);
-        $delegate
-            ->{HANDLER_METHOD}(Argument::that([$this->request, 'reveal']))
+        $handler = $this->prophesize(RequestHandlerInterface::class);
+        $handler
+            ->handle(Argument::that([$this->request, 'reveal']))
             ->will([$response, 'reveal']);
 
 
         $middleware = new ProblemDetailsMiddleware();
-        $result = $middleware->process($this->request->reveal(), $delegate->reveal());
+        $result = $middleware->process($this->request->reveal(), $handler->reveal());
 
         $this->assertSame($response->reveal(), $result);
     }
 
     /**
      * @dataProvider acceptHeaders
      */
-    public function testDelegateNotReturningResponseResultsInProblemDetails(string $accept) : void
```

Comment:
This test case is no longer possible, because we have return type with handlers, so TypeError will be thrown if handler is not returning ResponseInterface.